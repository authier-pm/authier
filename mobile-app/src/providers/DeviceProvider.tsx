import React, { createContext } from 'react'
import { useForceUpdate } from '../../useForceUpdate'
import { device, Device } from '../utils/Device'

export const DeviceContext = createContext<Device>({} as any)

export const DeviceProvider = ({ children }) => {
  const forceUpdate = useForceUpdate()

  if (!device.fireToken) {
    device.initializePromise.then(() => {
      forceUpdate()
    })
    return null
  }

  return (
    <DeviceContext.Provider value={device}>{children}</DeviceContext.Provider>
  )
}
