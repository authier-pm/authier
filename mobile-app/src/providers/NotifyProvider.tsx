import AsyncStorage from '@react-native-async-storage/async-storage'
import React, {
  createContext,
  useState,
  useEffect,
  Dispatch,
  SetStateAction
} from 'react'

export const NotifyContext = createContext<{
  notifies: Array<Notifi>
  setNotifies: Dispatch<SetStateAction<Notifi[]>>
}>({} as any)

export interface Notifi {
  pageName: string
  device: string
  icon: string
  location: string
  time: string
}

export default function NotifyProvider({ children }) {
  const [notifies, setNotifies] = useState<Notifi[]>([])

  useEffect(() => {
    const getData = async () => {
      const jsonValue = await AsyncStorage.getItem('notifies', (e) => {
        if (e) console.log(e)
      })

      if (jsonValue) {
        let data = JSON.parse(jsonValue as string)
        setNotifies([data.data])
      } else {
        setNotifies([])
      }
    }

    getData()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  return (
    <NotifyContext.Provider value={{ notifies, setNotifies }}>
      {children}
    </NotifyContext.Provider>
  )
}
