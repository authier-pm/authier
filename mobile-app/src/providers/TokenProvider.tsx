import React, { createContext, useState, useEffect } from 'react'
import messaging from '@react-native-firebase/messaging'

export const GetTokenProvider = createContext<{ token: string | null }>(
  {} as any
)

export default function TokenProvider({ children }) {
  const [token, setToken] = useState<string | null>(null)

  useEffect(() => {
    async function getToken() {
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const refresed = await messaging().onTokenRefresh(async (fcm) => {
        console.log('test')
        setToken(fcm)
        return
      })
      const Token = await messaging().getToken()
      console.log('t', Token)
      setToken(Token)
    }

    getToken()
  }, [])

  return (
    <GetTokenProvider.Provider value={{ token }}>
      {children}
    </GetTokenProvider.Provider>
  )
}
