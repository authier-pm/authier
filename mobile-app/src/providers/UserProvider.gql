query isLoggedIn {
  authenticated
}

mutation logout {
  logout
}
