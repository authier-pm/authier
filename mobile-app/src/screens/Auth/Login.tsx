import { Formik, FormikHelpers } from 'formik'
import {
  Heading,
  VStack,
  FormControl,
  Input,
  Button,
  Text,
  View,
  Pressable,
  HStack,
  Icon
} from 'native-base'
import React, {
  Dispatch,
  SetStateAction,
  useContext,
  useEffect,
  useState
} from 'react'

import Ionicons from 'react-native-vector-icons/Ionicons'
import { Trans } from '@lingui/macro'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import { AuthStackParamList } from '@navigation/AuthNavigation'
import { LoginAwaitingApproval } from './LoginAwaitingApproval'
import { Loading } from '@src/components/Loading'
import { useDeviceStore } from '@src/utils/deviceStore'

export interface ILoginFormValues {
  email: string
  password: string
  submitted: boolean
}

type NavigationProps = NativeStackScreenProps<AuthStackParamList, 'Login'>

export const LoginContext = React.createContext<{
  formState: ILoginFormValues
  setFormState: Dispatch<SetStateAction<ILoginFormValues>>
}>({} as any)

export function Login({ navigation }: NavigationProps) {
  const [loading, setLoading] = useState(false)
  const [show, setShow] = React.useState(false)
  const { formState, setFormState } = useContext(LoginContext)

  const device = useDeviceStore((state) => state)

  useEffect(() => {
    if (!device.fireToken) {
      setLoading(true)
    }
    setLoading(false)
  }, [device.fireToken])

  if (loading) {
    return <Loading />
  }

  if (formState.submitted) {
    return <LoginAwaitingApproval />
  }

  return (
    <View p="8" justifyContent="center">
      <Heading fontWeight="600">
        <Trans>Welcome</Trans>
      </Heading>
      <Heading mt="1" fontWeight="medium" size="xs">
        <Trans>Log in to continue!</Trans>
      </Heading>

      <Formik
        initialValues={formState}
        enableReinitialize
        onSubmit={async (
          values: ILoginFormValues,
          { setSubmitting }: FormikHelpers<ILoginFormValues>
        ) => {
          setFormState({
            ...values,
            submitted: true
          })
          setSubmitting(false)
        }}
      >
        {({
          handleChange,
          handleBlur,
          handleSubmit,
          values,
          isSubmitting,
          errors
        }) => {
          return (
            <VStack space={3} mt="5">
              <FormControl>
                <FormControl.Label
                  _text={{
                    fontSize: 'xl',
                    fontWeight: 500
                  }}
                >
                  Email
                </FormControl.Label>
                <Input
                  onChangeText={handleChange('email')}
                  onBlur={handleBlur('email')}
                  value={values.email}
                />

                <FormControl.ErrorMessage>
                  {errors.email}
                </FormControl.ErrorMessage>
              </FormControl>

              <FormControl>
                <FormControl.Label
                  _text={{
                    fontSize: 'xl',
                    fontWeight: 500
                  }}
                >
                  Password
                </FormControl.Label>
                <Input
                  onChangeText={handleChange('password')}
                  onBlur={handleBlur('password')}
                  value={values.password}
                  type={show ? 'text' : 'password'}
                  InputRightElement={
                    <Icon
                      as={
                        <Ionicons
                          name={show ? 'eye-outline' : 'eye-off-outline'}
                        />
                      }
                      size={5}
                      mr="2"
                      color="muted.400"
                      onPress={() => setShow(!show)}
                    />
                  }
                  placeholder="Password"
                />

                <FormControl.ErrorMessage>
                  {errors.password}
                </FormControl.ErrorMessage>
              </FormControl>

              <Button
                onPress={handleSubmit as (values: any) => void}
                isLoading={isSubmitting}
              >
                <Trans>Submit</Trans>
              </Button>
              <HStack mt="2" justifyContent="center" space={1}>
                <Text fontSize="sm" color="muted.500" fontWeight={400}>
                  <Trans>I'm a new user.</Trans>
                </Text>
                <Pressable
                  onPress={() =>
                    navigation.navigate('Register', {
                      password: values.password,
                      email: values.email
                    })
                  }
                >
                  <Text
                    color={'indigo.500'}
                    fontWeight={'medium'}
                    fontSize={'sm'}
                  >
                    <Trans>Sign Up</Trans>
                  </Text>
                </Pressable>
              </HStack>
              {/* <Button onPress={() => navigation.navigate('QRLogin')}>
              With QR Code
            </Button> */}
            </VStack>
          )
        }}
      </Formik>
    </View>
  )
}
