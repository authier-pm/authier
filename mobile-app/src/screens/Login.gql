mutation login($email: String!, $password: String!) {
  login(email: $email, password: $password) {
    accessToken
    user {
      id
      secrets {
        encrypted
        kind
        id
      }
    }
  }
}
