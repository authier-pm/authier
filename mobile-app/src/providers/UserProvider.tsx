import React, {
  createContext,
  useState,
  Dispatch,
  SetStateAction,
  useEffect
} from 'react'
import messaging from '@react-native-firebase/messaging'
import { useIsLoggedInQuery } from './UserProvider.codegen'
import { accessToken, clearAccessToken } from '../utils/tokenFromAsyncStorage'
import jwtDecode from 'jwt-decode'

export const UserContext = createContext<{
  isLogged: boolean
  setIsLogged: Dispatch<SetStateAction<boolean>>
  token: string | null
  isApiLoggedIn: Boolean
  logout: () => void
}>({} as any)

export default function UserProvider({ children }) {
  const [isLogged, setIsLogged] = useState(false)
  const [token, setToken] = useState<string | null>(null)
  const { data, loading } = useIsLoggedInQuery()

  useEffect(() => {
    async function getToken() {
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      messaging().onTokenRefresh(async (fcm) => {
        console.log('ressetting token')
        setToken(fcm)
        return
      })
      const Token = await messaging().getToken()
      setToken(Token)
    }

    //Check asyncStorage if is the accessToken valid
    async function isAccessTokenValid() {
      const token = accessToken

      if (!token) {
        setIsLogged(false)
        return false
      }

      try {
        // @ts-expect-error
        const { exp } = jwtDecode(token)
        if (Date.now() >= exp * 1000) {
          setIsLogged(false)
          return false
        } else {
          setIsLogged(true)
          return true
        }
      } catch (error) {
        return false
      }
    }

    isAccessTokenValid()
    getToken()
  }, [data, loading])

  return (
    <UserContext.Provider
      value={{
        isLogged,
        setIsLogged,
        token,
        isApiLoggedIn: !!(data?.authenticated && !loading),
        logout: async () => {
          setIsLogged(false)
          clearAccessToken()
          // await logout()
        }
      }}
    >
      {children}
    </UserContext.Provider>
  )
}
