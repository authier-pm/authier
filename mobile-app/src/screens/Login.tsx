import { Formik } from 'formik'
import {
  Heading,
  VStack,
  FormControl,
  Input,
  Button,
  Text,
  View,
  Pressable,
  HStack
} from 'native-base'
import React, { useContext } from 'react'

import SInfo from 'react-native-sensitive-info'
import { UserContext } from '../providers/UserProvider'
import { saveAccessToken } from '../../util/tokenFromAsyncStorage'

interface MyFormValues {
  email: string
  password: string
}

export function Login({ navigation }) {
  const initialValues: MyFormValues = { email: 'bob@bob.com', password: 'bob' }
  // const [login, { loading }] = useLoginMutation()
  const { setIsLogged } = useContext(UserContext)

  const saveData = async (value) => {
    try {
      SInfo.setItem('encryptedSecrets', value, {
        sharedPreferencesName: 'mySharedPrefs',
        keychainService: 'myKeychain'
      })
    } catch (error) {
      console.error(error)
    }
  }

  return (
    <View flex={1} p="2" w="90%" mx="auto" justifyContent="center">
      <Heading size="lg" fontWeight="600" color="coolGray.800">
        Welcome
      </Heading>
      <Heading mt="1" color="coolGray.600" fontWeight="medium" size="xs">
        Log in to continue!
      </Heading>

      <Formik
        initialValues={initialValues}
        onSubmit={async (values, actions) => {
          const response = await login({
            variables: { email: values.email, password: values.password }
          })

          if (response.data?.login?.accessToken) {
            let concat = response.data.login.user.EncryptedSecrets?.map((i) => {
              return i.encrypted
            })

            //save secrets
            saveData(concat ? JSON.stringify(concat) : JSON.stringify([]))
            saveAccessToken(response.data.login.accessToken)

            setIsLogged(true)
            actions.setSubmitting(false)
          } else {
            console.error(`Login failed, check your password`)
          }
          actions.setSubmitting(false)
        }}
      >
        {({ handleChange, handleBlur, handleSubmit, values }) => (
          <VStack space={3} mt="5">
            <FormControl>
              <FormControl.Label
                _text={{
                  color: 'coolGray.800',
                  fontSize: 'xl',
                  fontWeight: 500
                }}
              >
                Email
              </FormControl.Label>
              <Input
                onChangeText={handleChange('email')}
                onBlur={handleBlur('email')}
                value={values.email}
              />
            </FormControl>

            <FormControl>
              <FormControl.Label
                _text={{
                  color: 'coolGray.800',
                  fontSize: 'xl',
                  fontWeight: 500
                }}
              >
                Password
              </FormControl.Label>
              <Input
                onChangeText={handleChange('password')}
                onBlur={handleBlur('password')}
                value={values.password}
                type="password"
              />
              <Pressable alignSelf="flex-end" mt="1">
                <Text fontSize="xs" fontWeight="500" color="indigo.500">
                  Forget Password?
                </Text>
              </Pressable>
            </FormControl>

            <Button onPress={handleSubmit} isLoading={loading}>
              Submit
            </Button>
            <HStack mt="2" justifyContent="center">
              <Text fontSize="sm" color="muted.700" fontWeight={400}>
                I'm a new user.{' '}
              </Text>
              <Pressable onPress={() => navigation.navigate('Register')}>
                <Text
                  color={'indigo.500'}
                  fontWeight={'medium'}
                  fontSize={'sm'}
                >
                  Sign Up
                </Text>
              </Pressable>
            </HStack>
            <Button onPress={() => navigation.navigate('QRLogin')}>
              With QR Code
            </Button>
          </VStack>
        )}
      </Formik>
    </View>
  )
}
