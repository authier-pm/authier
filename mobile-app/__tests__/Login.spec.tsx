import React from 'react'
import { wrapInProviders } from '../test/providers'
import renderer from 'react-test-renderer'
import { Login } from '../src/screens/Login'

it('Home component renders', () => {
  const tree = renderer.create(wrapInProviders(<Login />)).toJSON()
  expect(tree).toMatchSnapshot()
})
