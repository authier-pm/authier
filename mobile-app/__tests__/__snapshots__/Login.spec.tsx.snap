// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    flex={1}
    justifyContent="center"
    mx="auto"
    p="2"
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "marginLeft": "auto",
          "marginRight": "auto",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "width": "90%",
        },
      ]
    }
    w="90%"
  >
    <Text
      color="coolGray.800"
      fontSize="2xl"
      fontStyle="normal"
      fontWeight="600"
      style={
        Array [
          Object {
            "color": "#1f2937",
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
    >
      Welcome to Authier
    </Text>
    <Text
      color="coolGray.600"
      fontSize="sm"
      fontStyle="normal"
      fontWeight="medium"
      mt="1"
      style={
        Array [
          Object {
            "color": "#4b5563",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "500",
            "marginTop": 4,
          },
        ]
      }
    >
      Log in to continue!
    </Text>
    <View
      mt="5"
      style={
        Array [
          Object {
            "marginTop": 20,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "width": "100%",
            },
          ]
        }
        width="100%"
      >
        <View
          flexDirection="row"
          justifyContent="flex-start"
          mb={2}
          mr={3}
          nativeID="field-react-aria-0-1-label"
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginBottom": 8,
                "marginRight": 12,
              },
            ]
          }
        >
          <Text
            color="coolGray.800"
            fontSize="xl"
            fontStyle="normal"
            fontWeight={500}
            style={
              Array [
                Object {
                  "color": "#1f2937",
                  "fontSize": 20,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                },
              ]
            }
          >
            Email ID
          </Text>
        </View>
        <TextInput
          accessible={true}
          allowFontScaling={true}
          background="transparent"
          borderColor="muted.200"
          borderRadius="lg"
          borderWidth={1}
          color="black"
          disabled={false}
          editable={true}
          fontSize="md"
          fontStyle="normal"
          fontWeight={400}
          isRequired={false}
          nativeID="field-react-aria-0-1"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          placeholderTextColor="#a3a3a3"
          px={4}
          py={3}
          readOnly={false}
          rejectResponderTermination={true}
          required={false}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderColor": "#e5e5e5",
                "borderRadius": 6,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "400",
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 12,
              },
            ]
          }
          underlineColorAndroid="transparent"
          variant="outline"
        />
      </View>
      <View
        height={3}
        style={
          Array [
            Object {
              "height": 12,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "width": "100%",
            },
          ]
        }
        width="100%"
      >
        <View
          flexDirection="row"
          justifyContent="flex-start"
          mb={2}
          mr={3}
          nativeID="field-react-aria-0-2-label"
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginBottom": 8,
                "marginRight": 12,
              },
            ]
          }
        >
          <Text
            color="coolGray.800"
            fontSize="xl"
            fontStyle="normal"
            fontWeight={500}
            style={
              Array [
                Object {
                  "color": "#1f2937",
                  "fontSize": 20,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                },
              ]
            }
          >
            Password
          </Text>
        </View>
        <TextInput
          accessible={true}
          allowFontScaling={true}
          background="transparent"
          borderColor="muted.200"
          borderRadius="lg"
          borderWidth={1}
          color="black"
          disabled={false}
          editable={true}
          fontSize="md"
          fontStyle="normal"
          fontWeight={400}
          isRequired={false}
          nativeID="field-react-aria-0-2"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          placeholderTextColor="#a3a3a3"
          px={4}
          py={3}
          readOnly={false}
          rejectResponderTermination={true}
          required={false}
          secureTextEntry={true}
          style={
            Array [
              Object {
                "borderColor": "#e5e5e5",
                "borderRadius": 6,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "400",
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 12,
              },
            ]
          }
          underlineColorAndroid="transparent"
          variant="outline"
        />
        <View
          mt="1"
          style={
            Array [
              Object {
                "marginTop": 4,
              },
            ]
          }
        >
          <View
            accessibilityRole="link"
            accessible={true}
            alignSelf="flex-end"
            collapsable={false}
            focusable={true}
            height="auto"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onHoverIn={[Function]}
            onHoverOut={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignSelf": "flex-end",
                  "height": "auto",
                  "width": "auto",
                },
              ]
            }
            width="auto"
          >
            <View
              alignSelf="flex-end"
              flexDirection="row"
              height="auto"
              style={
                Array [
                  Object {
                    "alignSelf": "flex-end",
                    "flexDirection": "row",
                    "height": "auto",
                    "width": "auto",
                  },
                ]
              }
              width="auto"
            >
              <Text
                color="indigo.500"
                fontSize="xs"
                fontStyle="normal"
                fontWeight="500"
                style={
                  Array [
                    Object {
                      "color": "#6366f1",
                      "fontSize": 12,
                      "fontStyle": "normal",
                      "fontWeight": "500",
                      "textDecorationLine": "none",
                    },
                  ]
                }
                textDecorationLine="none"
              >
                Forget Password?
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        height={3}
        style={
          Array [
            Object {
              "height": 12,
            },
          ]
        }
      />
      <View
        _text={
          Object {
            "color": "lightText",
          }
        }
        accessibilityRole="button"
        accessible={true}
        alignItems="center"
        bg="indigo.500"
        borderRadius="lg"
        collapsable={false}
        colorScheme="indigo"
        flexDirection="row"
        focusable={true}
        justifyContent="center"
        mt="2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        px={4}
        py={3}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#6366f1",
              "borderRadius": 6,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 8,
              "paddingBottom": 12,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 12,
            },
          ]
        }
        variant="solid"
      >
        <View
          alignItems="center"
          flexDirection="row"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              color="white"
              fontSize="md"
              fontStyle="normal"
              fontWeight={600}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "600",
                  },
                ]
              }
            >
              Log in
            </Text>
          </View>
        </View>
      </View>
      <View
        height={3}
        style={
          Array [
            Object {
              "height": 12,
            },
          ]
        }
      />
      <View
        flexDirection="row"
        justifyContent="center"
        mt="6"
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 24,
            },
          ]
        }
      >
        <Text
          color="muted.700"
          fontSize="sm"
          fontStyle="normal"
          fontWeight={400}
          style={
            Array [
              Object {
                "color": "#404040",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
              },
            ]
          }
        >
          I'm a new user.
           
        </Text>
        <View
          style={
            Array [
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <View
            accessibilityRole="link"
            accessible={true}
            collapsable={false}
            focusable={true}
            height="auto"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onHoverIn={[Function]}
            onHoverOut={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "height": "auto",
                  "width": "auto",
                },
              ]
            }
            width="auto"
          >
            <View
              flexDirection="row"
              height="auto"
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "height": "auto",
                    "width": "auto",
                  },
                ]
              }
              width="auto"
            >
              <Text
                color="indigo.500"
                fontSize="sm"
                fontStyle="normal"
                fontWeight="medium"
                style={
                  Array [
                    Object {
                      "color": "#6366f1",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "500",
                      "textDecorationLine": "none",
                    },
                  ]
                }
                textDecorationLine="none"
              >
                Sign Up
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
